linode/manager
# change: [M3-8290] - Improve experience of closing out of Support Ticket Dialog

## Description 📝
Currently, when a user closes (but not cancels) out of the support ticket dialog, their ticket title and description are saved to local storage. No other field values are intentionally saved.

To provide a consistent experience, this PR saves all of the *general* support ticket form in local storage if a user closes out the dialog and repopulates them upon reopening the dialog. Currently, there is no way to close out of and then reopen a special support ticket type (`smtp` or `accountLimit`), since only a `general` support ticket can be created with the Open New Ticket button.

Additionally, we tag the Cancel and Close buttons with analytics to see which users are making more use of. The discrepancy in behavior between Cancel (clear form) and Close (store and reload data) behavior is intentional, but may be odd/unexpected, and we can consider improving the UX in the future.

## Changes  🔄
- Updates local storage to store additional fields for general support tickets.
- Tags the `Cancel` and `X` buttons with analytics event.
- Fixes a edge case bug where opening a general support ticket after starting, but *closing out of* an account limit support ticket (not submitting) would pre-populate the account limit support ticket title, instead of clearing that value for the new general ticket.

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/user-attachments/assets/a485563f-a985-42b9-81ef-cfda56b95ad3" /> | <video src="https://github.com/user-attachments/assets/2c167626-87ac-45a6-989c-e898956316e7" /> |
| ![Screenshot 2024-07-23 at 1 05 30 PM](https://github.com/user-attachments/assets/9c3fc317-9a7c-4ba5-b6af-22447c1cc86e) | ![Screenshot 2024-07-23 at 12 52 40 PM](https://github.com/user-attachments/assets/86df2e74-578e-4e20-9647-da6b023ae5e9) |


## How to test 🧪

### Prerequisites
(How to setup test environment)
- **Please test this on your dev account so we don't bother Customer Support with real tickets.**
- Turn the Support Ticket Severity flag on in devtools.

### Reproduction steps
(How to reproduce the issue, if applicable)
- Go to [cloud.linode.com/support/tickets](https://cloud.linode.com/support/tickets), fill out all fields of the support ticket form, including severity and an entity selection, and then hit the close (x) button.
- Click the "Open New Ticket" button and observe that although the title and description are restored from data saved in local storage, the other values have been cleared.

### Verification steps
(How to verify changes)
Please verify:
- All general support ticket form values are stored in local storage when the user closes out of the dialog
- These values are then repopulated when the dialog is reopened
- There are no regressions in the Cancel behavior (all form fields are cleared) when this button is clicked
- There are no crashes – we've fixed the issue where creating (and successfully submitting) one support ticket and then attempting to create another would crash the app
- Analytics events are tagged on the close and cancel buttons
- There are no regressions when viewing and then closing out of the form (or actually creating) support tickets of various types (`smtp`, `accountLimit`, `general`).

> [!IMPORTANT]
> Please let me know if you see any crashes or white screens. In working with this PR, I would occasionally see the screen go white and a console failure about loading the splashscreen. I couldn't reliably replicate it and assume it was related to local storage updates while I was making changes, but want to be sure it's not still happening now.

## As an Author I have considered 🤔

*Check all that apply*
